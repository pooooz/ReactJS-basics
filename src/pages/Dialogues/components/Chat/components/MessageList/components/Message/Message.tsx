import React, { FC } from 'react';

import styles from './Message.module.scss';

interface FuncMessageProps {
  message: string;
  author: string;
}

export const Message: FC<FuncMessageProps> = (props) => {
  return (
    <div className={styles.message}>
      <p className={styles.glitch} data-text={props.message}>
        {props.message}
      </p>
      <p className={styles.glitch} data-text={'Author: <' + props.author + '>'}>
        Author: &lt;{props.author}&gt;
      </p>
    </div>
  );
};
