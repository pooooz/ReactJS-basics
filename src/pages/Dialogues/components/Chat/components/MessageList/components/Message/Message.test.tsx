import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Message } from './Message';

describe('Message', () => {
  it('Render component', () => {
    render(<Message message="Test" author="Test" />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(<Message message="Test" author="Test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render component with author', () => {
    render(<Message message="Test" author="Test" />);
    expect(screen.getByText(/Author: <\w*>/)).toBeInTheDocument();
  });
});
