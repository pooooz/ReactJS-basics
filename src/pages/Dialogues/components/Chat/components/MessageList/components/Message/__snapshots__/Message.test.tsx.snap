// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuncMessage Render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      data-text="Test"
    >
      Test
    </p>
    <p
      data-text="Author: <Test>"
    >
      Author: &lt;Test&gt;
    </p>
  </div>
</DocumentFragment>
`;
